src/sysutil.rs

```diff
@@ -224,31 +283,18 @@ impl SysUtil {
                     name_regex.captures(text).and_then(|capture| capture.get(1))
                 {
                     debug!(
-                        "Found name '{}' for session: {}",
+                        "Found name '{}' for session '{}' with command '{:?}'",
                         name.as_str(),
-                        path.display()
+                        path.display(),
+                        cmd
                     );
                     name.as_str()
-                } else if let Some(stem) = path.file_stem() {
-                    // Get the stem of the filename of this desktop file.
-                    // This is used as backup, in case the file name doesn't exist.
-                    if let Some(stem) = stem.to_str() {
-                        debug!(
-                            "Using file stem '{stem}', since no name was found for session: {}",
-                            path.display()
-                        );
-                        stem
-                    } else {
-                        warn!("Non-UTF-8 file stem in session file: {}", path.display());
-                        // No way to display this session name, so just skip it.
-                        continue;
-                    }
                 } else {
-                    warn!("No file stem found for session: {}", path.display());
-                    // No file stem implies no file name, which shouldn't happen.
-                    // Since there's no full name nor file stem, just skip this anomalous
-                    // session.
-                    continue;
+                    debug!(
+                        "Using file stem '{stem}', since no name was found for session: {}",
+                        path.display()
+                    );
+                    &stem
```

Comment:
If you do remove the UTF-8 requirement for `stem` above, you would have to add it here.

Reply:
Will do